spec/jobs/place_holds_job_spec.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PlaceHoldsJob, type: :job do
+  let(:ws_args) { {
+    patron_key: 'patron1',
+    barcodes: ['1'],
+    session_token: 'session_token',
+    pickup_library: 'UP-PAT',
+    pickup_by_date: '01-20-2021'
+  } }
+
+  after do
+    Redis.current.flushall
+  end
+
+  context 'with valid input' do
+    before do
+      stub_request(:any, /example.com/).to_rack(FakeSymphony)
+    end
+
+    it 'sets a Redis record containing success denoted by patron\'s key' do
+      described_class.perform_now(**ws_args)
+      results = Redis.current.get 'place_holds_results_patron1'
+
+      expect(results).to be_present
+    end
+
+    it 'renders HTML containing the checkouts and saves to redis' do
+      described_class.perform_now(**ws_args)
+      results = Redis.current.get 'place_holds_results_patron1'
+
+      expect(results).to include '1 Hold Placed'
+    end
```

Comment:
do we need the first spec? isn't the latter also testing for that case? 🤔  I might be missing sth as well.

Reply:
I think the idea is sort of like testing baby steps, but yes, the seconds spec implicitly covers the first. But there could be a scenario in which the first spec passes and the second doesn't. As in, something is wrong with rendering piece but not necessarily the Redis piece. I also was just following the pattern we used in other Job specs, like https://github.com/psu-libraries/myaccount/blob/master/spec/jobs/view_checkouts_job_spec.rb#L18-L29

Reply:
For the case when second fails and first passes, it tells us yeah this exists but not much other than that, right? I am OK to leave them but I think they are not adding much.

Reply:
In the case of the first spec passing it is a simple test of `Redis.current.set` worked and was able to use `@patron.key` (which in itself is a SymphonyClient call and mapping to a new instance of Patron). So, I think that is a fair amount worth testing, I don't know.

Then aside, we want to make sure this part worked https://github.com/psu-libraries/myaccount/pull/213/files?file-filters%5B%5D=.rb#diff-0bb303924e2704cb7e53481994a126e6R25-R33. Which is some more complexity - service classes and views called on. 

I wouldn't mind keeping both specs and I wouldn't mind deleting the first spec either, but if we do that, we should delete it from the others too, right?

Reply:
right, if we delete this we should delete the others, I kinda like that but I am also getting eager to merge this PR as well to move forward. It is all good, we can think about them later if we need to.